prisma/migrations/20250224065512_session_trigger/migration.sql

```diff
@@ -16,9 +16,26 @@ LANGUAGE plpgsql
 SECURITY DEFINER
 SET search_path = public
 AS $$
+DECLARE
+    country_code VARCHAR(2);
 BEGIN
-    INSERT INTO "Sessions" (id, "createdAt", "updatedAt", ip, "userAgent", "userId", "deviceNonce")
-    SELECT NEW.id, NEW.created_at, NEW.updated_at, NEW.ip, NEW.user_agent, NEW.user_id, gen_random_uuid();
+    SELECT "countryCode" INTO country_code from "IpGeolocation" where ip = NEW.ip;
+
+    INSERT INTO "Sessions" (id, "createdAt", "updatedAt", ip, "userAgent", "userId", "deviceNonce", "countryCode")
+    SELECT NEW.id, NEW.created_at, NEW.updated_at, NEW.ip, NEW.user_agent, NEW.user_id, gen_random_uuid(), COALESCE(country_code, '');
+
+    -- If country code wasn't in cache, fetch it asynchronously
+    IF country_code IS NULL THEN
+        BEGIN
+            PERFORM net.http_get('https://freeipapi.com/api/json/' || host(NEW.ip));
```

Comment:
I think we might get rate-limitted eventually. Because the country code is redundant (as it links one-to-one with the IP), maybe we can just remove it from the challenge data. This way we avoid using this API and having to wait for this async data to "complete" the session.

Reply:
I think we can remove it if its redundant and not so useful as we already have `ip` but we can already get the `countryCode` from the request in the trpc context where it updates the session with it. https://github.com/wanderwallet/embed-api/blob/de120e02b58ccaa7a1ad224695b47d4f618fe0b8/server/context.ts#L46
This PR is just so that the session has the `countryCode` when it is created if there is `ip` and `countryCode` already present in cache table.

Should we remove it wdyt?

Reply:
Yes, I think it's better to remove it from the challenges and use it only to display it to users (eventually). I'd say we can even remove the field from `Session`, as once we want to show it to users, we can create a separate entity/endpoint that maps IPs to countries and only adds/updates values when they are going to be consumed.

Reply:
Sounds good. Will update this PR and will also create a PR for frontend as well.